### DISCUSSION TITLE
v1 - Performance 📈

### DISCUSSION THREAD

**Discussion by alisonjoseph** [carbon-design-system/carbon-platform#1273] [Release feedback]
**Hi 👋  Thanks for joining our v1 "Test Fest"!** 

Are you noticing any performance issues? Please log them here. We really appreciate your help ❤️ 

```
📈 Performance issue
**Page:** URL
**Browser:** 
**Issue:** 
**Suggestion:** 
```

**Comment by alisonjoseph** [comment]
📈 Performance issue
**Page:** https://next.carbondesignsystem.com/libraries/carbon-components-angular
**Browser:** ALL
**Issue:** Navigating from `/libraries` into any library is extremely slow, 20+ seconds for the page to to open for me. 
**Suggestion:**

**Reply by mattrosno** [reply]
I think we're feeling the `fallback: 'blocking'` that we're using in all of our ISR pages:
<img width="1276" alt="image" src="https://user-images.githubusercontent.com/1691245/190698045-199da5ed-6612-4b5f-898c-0d04ca94f854.png">

If we use `fallback: true` on all of these, then in each page, we can do this at the top of each page's default component:

```jsx
const router = useRouter();

if (router.isFallback) {
  return <Loading />;
}
```

Where we render a loading component for the full page, so the user can see that the page is getting built. That will help so the page immediately goes to the new route when you click a page link that hasn't been built yet.

**Reply by mattrosno** [reply]
Ideally, we'd do a loading page per each page type with page structure, skeletons, etc. Initially, we could probably get by with a full-page loading indicator: https://next.carbondesignsystem.com/libraries/carbon-styles/latest/assets/loading-pattern/usage#full-screen-loading

**Reply by mattrosno** [reply]
>If a loading process will require the application’s full resources and will take longer than a moment or two, use a full-screen loading indicator.

**Reply by jharvey10** [reply]
Unfortunately `fallback: true` does not work with ISR :(

https://nextjs.org/docs/api-reference/data-fetching/get-static-paths#fallback-true

<img width="713" alt="image" src="https://user-images.githubusercontent.com/51208233/190704761-dd6f311d-ec10-4bd3-a997-65565d9f0276.png">

**Reply by mattrosno** [reply]
@jdharvey-ibm those docs are confusing. Per other examples, it appears you can use `fallback: true` with ISR. See: https://www.smashingmagazine.com/2021/04/incremental-static-regeneration-nextjs/#generating-paths

https://www.smashingmagazine.com/2021/04/incremental-static-regeneration-nextjs/#isr-fallback-options

<img width="681" alt="image" src="https://user-images.githubusercontent.com/1691245/190709869-0b8df4e4-439f-4fa4-a861-b010251f7001.png">

**Reply by jharvey10** [reply]
Oh interesting! Yeah the nextjs docs definitely don't make that clear. We could definitely try out `fallback: true` for pages for which we don't care about SEO.

**Reply by mattrosno** [reply]
@jdharvey-ibm per https://nextjs.org/docs/api-reference/data-fetching/get-static-paths#fallback-true

>Web crawlers, such as Google, won't be served a fallback and instead the path will behave as in [fallback: 'blocking'](https://nextjs.org/docs/api-reference/data-fetching/get-static-paths#fallback-blocking).

I think Next.js is smart enough to know if the request is from a crawler, to not show the fallback or any loader rendered when no fallback is available, and instead wait on the SSR (`fallback: 'blocking'`) page.

**Reply by jharvey10** [reply]
Well that's pretty cool, haha. Takes a lot off of our plate then!

**Reply by mattrosno** [reply]
Tracked here: https://github.com/carbon-design-system/carbon-platform/issues/1313
